model/project_parser_db.go

```diff
@@ -73,15 +76,21 @@ func (pp *ParserProject) UpsertWithConfigNumber(num int) error {
 	}
 	if found == nil {
 		pp.ConfigUpdateNumber = num
+		// this could error but it seems unlikely
```

Comment:
What would happen if it did? The network could drop or there could be a failover, for example. Would that have some far-reaching effect on a single version?

Reply:
It's okay if it errors and it should, specifically the error I would be concerned about would be the duplicate key error (i.e. if the document was inserted between the time of lookup and insert). What does generate tasks do in this case? Potentially if it's a duplicate key error we shouldn't return an error and should just continue to the update.

Reply:
Before your change, if the version.config updated errored, the generator would abort, and eventually another generator would try to do the work again. You can safely error a job, because another generator will get created if one doesn't succeed.

Reply:
Ah actually this shouldn't error then I think--because it's the "force update parser project" step, the version will already have been updated, and another generator trying to do the work again will cause an error. I think in the case of an error we should log and then also try to update.